### PR TITLE
Add tags app

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: dash/tags/models.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class Tag(SmartModel):
+
+    name = models.CharField(max_length=64, help_text=_("The name of this tag"))
+    org = models.ForeignKey(Org, on_delete=models.PROTECT, help_text=_("The organization this tag belongs to"))
+
+    def __str__(self) -> str:
+        return f"{self.org} - {self.name}"
+
+    class Meta:
+        unique_together = ("name", "org")
```
[the remark this answers is not in the document]
Comment (norkans7) [reply dash/tags/models.py:18]: Added the delete view

File: dash/tags/models.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from dash.orgs.models import Org
+
+
+class Tag(SmartModel):
+
+    name = models.CharField(max_length=64, help_text=_("The name of this tag"))
+    org = models.ForeignKey(Org, on_delete=models.PROTECT)
+
+    def __str__(self) -> str:
+        return f"{self.name}"
```
Comment (rowanseymour) [inline dash/tags/models.py:15]: `return self.name`

File: dash/tags/views.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    class Create(OrgPermsMixin, SmartCreateView):
+        fields = ("name",)
+
+        def pre_save(self, obj):
+            obj = super(TagCRUDL.Create, self).pre_save(obj)
+
+            org = self.derive_org()
+            if org:
+                obj.org = org
```
Comment (rowanseymour) [inline dash/tags/views.py:22]: I don't think `org` can be null, no?

File: dash/tags/views.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+            return obj
+
+    class Update(OrgObjPermsMixin, SmartUpdateView):
+        fields = ("name",)
+        delete_url = "id@tags.tag_delete"
+
+    class List(OrgPermsMixin, SmartListView):
+        fields = ("name", "modified_on", "created_on")
+        ordering = ("name",)
```
Comment (rowanseymour) [inline dash/tags/views.py:32]: this is ignored as well as the `ordering` attribute on the model if you're explicitly ordering the queryset by `Lower("name")`

File: pyproject.toml [hunk 5]
```diff
@@ -37,10 +37,13 @@ smartmin = "^3.0.0"
 sorl-thumbnail = "^12.7.0"
 
 [tool.poetry.dev-dependencies]
-black = "^20.8b1"
+black = [
+    { version = "^20.8b1", python = "^3.6.0", allow-prereleases = true},
+    { version = "^21.7b0", python = ">3.7.0", allow-prereleases = true}
+]
```
Comment (rowanseymour) [inline pyproject.toml:43]: does latest black not support python 3.6?

Comment (norkans7) [reply pyproject.toml:43]: Yes, https://github.com/rapidpro/dash/runs/3499113420

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**